### DISCUSSION TITLE
How do you add composer auth and custom git keys for private repo's?

### DISCUSSION THREAD

**Discussion by leroy0211** [dunglas/symfony-docker#615] [Q&A] [answered]
I would like to know how you would get a custom composer `auth.json` or deployment keys into the image. 

We have a few private composer packages we need to include in our projects, and either need a deployment key or some other form of ssh keys.  

It's OK if we limit this to development only, because during CI/CD builds we use a form of ENV keys to specify the deployment keys. But this doesn't work during development, since each developer uses their own key.

**Comment by dunglas** [comment, marked as answer]
See my article about this topic: https://dunglas.dev/2022/08/securely-access-private-git-repositories-and-composer-packages-in-docker-builds/

**Reply by leroy0211** [reply]
Your link describes the production build. But I'm very interesting in a local development.  

This project doesn't install composer packages while targeting the development build, which is great because you will probably mount your `vendor/` directory from your host. 
However we would like to run `composer` from within the container so we don't get the issue "it works on my machine" because of a php version mismatch.  

So when running `composer install` as `docker compose exec composer install` you don't get the benefits of ssh forwarding, nor any of your composer auth.json config locally stored on your development host.

**Reply by 7-zete-7** [reply]
Hello @leroy0211!

On the go, when working with a container, it will be enough to mount the `$HOME/.composer/auth.json` or the `$SSH_AUTH_SOCK` binds into your container.

When building, you will need to connect the `$HOME/.composer/auth.json` or SSH Agent socket through the methods described in the article that @dunglas suggested.

To run this command on a remote server, it would be easier to create a temporary token for Composer and remove it after the command is completed.
